icevision/models/torchvision_models/faster_rcnn/show_results.py

```diff
@@ -31,3 +36,103 @@ def show_results(
         denormalize_fn=denormalize_fn,
         show=show,
     )
+
+
+def get_losses(
+    model: nn.Module,
+    dataset: Dataset,
+) -> Tuple[List[dict], dict]:
+    """
+    Gets a dataset and a model as input and returns losses calculated per image + losses stats.
+    In case of faster_rcnn those are loss_classifier, loss_box_reg and loss_total
+    (sum of the first 2 plus RPN related losses, e.g. sum of 4 terms).
+
+    Arguments
+    ---------
+    model: nn.Module
+    dataset: Dataset
+
+    Returns
+    -------
+    samples_plus_losses: those are the samples (dictionaries) contained in the original dataset
+                          enriched with losses.
+    losses_stats: dictionary containing losses stats (min, max, mean, 1-25-50-75-99 quantiles)
+                  for each one of the losses.
+    """
+    device = torch.device("cpu")
+    model = model.to(device)
+    dl = valid_dl(dataset, batch_size=1, num_workers=0, shuffle=False)
+
+    torch.manual_seed(0)
```

Comment:
If we want all samples to have the same dropout values and all, should we move the manual seed to inside the for loop?

Actually, should we try to find a way to put the model in eval model while still having it to calculate the loss?

Inside torchvision in `generalized_rcnn.py` we see that the forward loop checks for the variable `self.training`, what happens if we do:
```python
model.eval()
model.training = True
```

Reply:
> If we want all samples to have the same dropout values and all, should we move the manual seed to inside the for loop?

I checked and in-outside the loop does not change the result. The seed is set properly.

> what happens if we do:

I can try doing that indeed!

Reply:
This is interesting, to say the least! Not sure what is going on...

![image](https://user-images.githubusercontent.com/1206519/102224952-e75d8c80-3ee6-11eb-8216-c3627f575de7.png)

Reply:
So this works:
```
model.eval()
model.train()
model(x, y)
```
e.g. it does not return `{}` but proper losses, but it does not solve the problem of the seed.
We can totally use this approach to avoid copying the model (see 2 comments below) but the seed will still be needed.

Reply:
Done, moved the `seed` to inside the loop.